docs/03_server/04_event-handler/03_advanced.md

```diff
@@ -241,110 +251,247 @@ eventHandler {
 
+`approvableAck` can be used in both custom EventHandler definitions and GPAL event handlers. Here is an example of `approvableAck` in action example for a custom event handler `onValidate` block below.
 
 ```kotlin
     override suspend fun onValidate(message: Event<Company>): EventReply {
         val company = message.details
-        // custom code block..
-        return ack(listOf(mapOf("ENTITY_TABLE" to "COMPANY", "ENTITY_ID" to company.companyId)))
+        // custom validation code block..
+        return approvableAck(
+            entityDetails = listOf(
+                ApprovalEntityDetails(
+                    entityTable = "COMPANY",
+                    entityId = event.details.companyId.toString(),
+                    approvalType = ApprovalType.UPDATE
+                )
+            ),
+            approvalMessage = "Company update for ${event.details.companyId} has been sent for approval.",
+            approvalType = ApprovalType.UPDATE,
+            additionalDetails = "Sensitive update, tread carefully"    
+        )
     }
 ```
 
-
-In order to display pending approvals against the original record in the GUI, you can use the `APPROVAL_ENTITY` table to join to the `APPROVAL_ENTITY` records in a view. You can then display the information using a Data Server or Request Server. The details of the pending event are stored in json format.
+Pending approval information is provided by the dataserver queries named `ALL_APPROVAL_ALERTS` and `ALL_APPROVAL_ALERTS_AUDITS`
 
 
 **Example APPROVAL DB record**
 
 
-```DbM
-TIMESTAMP          2018-09-19 09:29:51.111951844              NANO_TIMESTAMP
-APPROVAL_ID        000000000000002APLO1                       STRING
-APPROVAL_KEY       8a178f41-24c6-4cb3-b4e0-1996ae59bcddA...   STRING
-APPROVAL_MESSAGE   Please approve this amendment              STRING
-APPROVAL_STATUS    PENDING                                    ENUM[PENDING APPROVED CANCELLED REJECTED_BY_USER REJECTED_BY_SERVICE]
-DESTINATION        EEP_INTENT_SERVICE                         STRING
-MESSAGE_TYPE       EVENT_OPS_INTENT                           STRING
-EVENT_DETAILS      TRADE_ID = LCH20180917.18500000098 DE...   STRING
-EVENT_MESSAGE      {"MESSAGE_TYPE":"EVENT_OPS_INTENT","VAL... STRING
-USER_NAME          CdsTest                                    STRING
 ```
+-------------------------------------------------------------------------------------------
+TIMESTAMP                                2023-02-27 15:33:42.364(n:0,s:1019)      NANO_TIMESTAMP      
+ACTIONED_BY                              JaneDee                                  STRING              
+APPROVAL_ID                              fdef7802-6bd1-4c51-a232-6a4bc2325598A... STRING              
```

Comment:
Should we use sequence as if it were FDB?

Reply:
Not sure it matters too much, it just needs to show that the APPROVAL_ID here matches the one in the APPROVAL_ENTITY table below. I can change it for FDB if you think it makes more sense.